pkg/endpoints/common.go

```diff
@@ -16,3 +19,61 @@ func HttpNotOk(statusCode int, w http.ResponseWriter, frontendErr string, err er
 }
 
 const MaxUploadSize = 2 * 1024 * 1024
+
+func GetInt(r *http.Request, key string) (int, bool) {
+	var i int64 = 0
+	value, ok := r.URL.Query()[key]
+
+	if !ok {
+		return int(i), ok
+	}
+
+	i, err := strconv.ParseInt(value[0], 10, 64)
+
+	if err != nil {
+		return int(i), false
+	}
+
+	return int(i), true
+}
+
+func GetIntOk(r *http.Request, w http.ResponseWriter, key, message string) (int, bool) {
+	val, ok := GetInt(r, key)
+
+	if !ok {
+		err := errors.New(message)
+		HttpNotOk(400, w, err.Error(), err)
+	}
+
+	return val, ok
+}
+
+func GetStringOk(r *http.Request, w http.ResponseWriter, key, message string) (string, bool) {
+	value, ok := r.URL.Query()[key]
+	if !ok || len(value) < 1 {
+		if !ok {
+			err := errors.New("message")
+			HttpNotOk(400, w, err.Error(), err)
+		}
+	}
```

Comment:
check for the length in the outer if but then ignore it in the inner if. Inner check makes the outer check useless?